src/cuda/kernels/reorder.f90

```diff
@@ -199,6 +199,27 @@ attributes(global) subroutine axpby(n, alpha, x, beta, y)
 
    end subroutine axpby
 
+   attributes(global) subroutine scalar_product(s, x, y, n)
```

Comment:
Caller needs to understand this is pencil sum to set correct n_threads/blocks. This should be documented here.